joi-validations/topics.validations.js

```diff
@@ -1,11 +1,17 @@
-const {string} = require('./general.validations');
+const {string, url} = require('./general.validations');
 
 const TopicValidation = {
   latestPost: {
     query: {
       name: string.required()
     }
   },
+  updateTopic: {
+    body: {
+      icon: url,
+      cover: url
+    }
+  },
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `updateTopic` validation schema does not make the `icon` and `cover` fields required. If these fields are necessary for updating a topic, they should be marked as required. Otherwise, an update request could pass validation without these fields, leading to potential issues when trying to update the database.

```diff
  updateTopic: {
    body: {
-      icon: url,
-      cover: url
+      icon: url.required(),
+      cover: url.required()
    }
  },
```
If the `icon` and `cover` fields are optional, then this code is correct as it is.


<!-- This is an auto-generated comment by OSS CodeRabbit -->